posthog/api/test/test_survey.py

```diff
@@ -1109,6 +1109,68 @@ def test_update_basic_survey_question_validation(self):
         }
         assert response_data["created_by"]["id"] == self.user.id
 
+    def test_validate_link_url_scheme(self):
+        basic_survey = self.client.post(
+            f"/api/projects/{self.team.id}/surveys/",
+            data={
+                "name": "survey without targeting",
+                "type": "popover",
+            },
+            format="json",
+        ).json()
+
+        response = self.client.patch(
+            f"/api/projects/{self.team.id}/surveys/{basic_survey['id']}/",
+            data={
+                "name": "Notebooks beta release survey",
+                "description": "Get feedback on the new notebooks feature",
+                "type": "popover",
+                "questions": [
+                    {
+                        "type": "link",
+                        "link": "javascript:alert(1)",
+                        "question": "<b>What</b> do you think of the new notebooks feature?",
+                    },
+                ],
+            },
+            format="json",
+        )
+        invalid_url = "Link must be a URL to a https resource"
+
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+        assert response.json()["detail"] == invalid_url
+
+    def test_validate_link_url_location(self):
+        basic_survey = self.client.post(
```

Comment:
minor nit, but in both cases you seem to be testing on updates, would be nice to switch one of these to test validation on creation as well

Reply:
sounds good, I added tests for CREATE too and now allow mailto as a valid url scheme.